framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -2040,9 +2035,44 @@ public Void visitNewClass(NewClassTree node, Void p) {
         AnnotatedExecutableType constructorType = fromUse.executableType;
         List<AnnotatedTypeMirror> typeargs = fromUse.typeArgs;
 
+        // Type check inner class receiver type
+        AnnotatedTypeMirror parameterReceiverType = constructorType.getReceiverType();
+        AnnotatedTypeMirror argumentReceiverType = null;
+        if (node.getEnclosingExpression() != null) {
+            argumentReceiverType = atypeFactory.getAnnotatedType(node.getEnclosingExpression());
+        } else {
+            argumentReceiverType = atypeFactory.getImplicitReceiverType(node);
+        }
+        if (checker.hasOption("conservativeInnerClassReceiverCheck")
```

Comment:
1. Why is this check for the option only here and not the first thing of this change? Why determine the receiver types at all if they are not checked?